pkg/apis/core/v1beta1/helper/shoot.go

```diff
@@ -665,3 +666,28 @@ func IsUpdateStrategyInPlace(updateStrategy *gardencorev1beta1.MachineUpdateStra
 
 	return *updateStrategy == gardencorev1beta1.AutoInPlaceUpdate || *updateStrategy == gardencorev1beta1.ManualInPlaceUpdate
 }
+
+// GetMinAllowedForKubeAPIServer returns the minAllowed configuration values for Kube API Server if configured or nil otherwise.
+func GetMinAllowedForKubeAPIServer(apiServerConfig *gardencorev1beta1.KubeAPIServerConfig) corev1.ResourceList {
+	if apiServerConfig != nil && apiServerConfig.Autoscaling != nil {
+		return maps.Clone(apiServerConfig.Autoscaling.MinAllowed)
+	}
+	return nil
+}
```

Comment:
Do we really need to make this an API helper? Can't the KAPI component itself clone if it wants to inject additional values?

Reply:
It is surprising that we don't provide such "copy" functions for other maps we're having in our APIs. It rather seems like this is needed because of some component-specific logic because it might want to modify the maps' content. Hence, this better fits to the component package itself instead of the general API helper package, no?

Reply:
Removed the helpers and dropped the cloning.